corehq/apps/userreports/reports/view.py

```diff
@@ -668,18 +664,27 @@ def section_url(self):
         return "#"
 
     def get(self, request, *args, **kwargs):
-        context = super(DownloadUCRStatusView, self).main_context
-        context.update({
-            'domain': self.domain,
-            'download_id': kwargs['download_id'],
-            'poll_url': reverse('ucr_download_job_poll', args=[self.domain, kwargs['download_id']]),
-            'title': _("Download Report Status"),
-            'progress_text': _("Preparing report download."),
-            'error_text': _("There was an unexpected error! Please try again or report an issue."),
-            'next_url': reverse(ConfigurableReport.slug, args=[self.domain, self.report_config_id]),
-            'next_url_text': _("Go back to report"),
-        })
-        return render(request, 'hqwebapp/soil_status_full.html', context)
+        if _has_permission(self.domain, request.couch_user, self.report_config_id):
+            context = super(DownloadUCRStatusView, self).main_context
+            context.update({
+                'domain': self.domain,
+                'download_id': kwargs['download_id'],
+                'poll_url': reverse('ucr_download_job_poll',
+                                    args=[self.domain, kwargs['download_id']],
+                                    params={'config_id': self.report_config_id}),
+                'title': _("Download Report Status"),
+                'progress_text': _("Preparing report download."),
+                'error_text': _("There was an unexpected error! Please try again or report an issue."),
+                'next_url': reverse(ConfigurableReport.slug, args=[self.domain, self.report_config_id]),
+                'next_url_text': _("Go back to report"),
+            })
+            return render(request, 'hqwebapp/soil_status_full.html', context)
+        else:
+            raise Http403()
```

Comment:
totally a style thing, but I'd do this at the top of the function - `if not has_permission: raise`, that way it's a special case that escapes early rather than a fork in the logic.  Same with the other one down below.